src/pgduckdb_utils.cpp

```diff
@@ -0,0 +1,63 @@
+#include "duckdb.hpp"
+
+extern "C" {
+#include "postgres.h"
+}
+
+#include "pgduckdb/pgduckdb_utils.hpp"
+
+namespace pgduckdb {
+
+template <typename T, typename FuncType, typename... FuncArgs>
+T
+PostgresFunctionGuard(FuncType postgres_function, FuncArgs... args) {
+	T return_value;
+	bool error = false;
+	MemoryContext ctx = CurrentMemoryContext;
+	ErrorData *edata = nullptr;
+	// clang-format off
+	PG_TRY();
+	{
+		return_value = postgres_function(args...);
+	}
+	PG_CATCH();
+	{
+		MemoryContextSwitchTo(ctx);
+		edata = CopyErrorData();
+		FlushErrorState();
+		error = true;
+	}
+	PG_END_TRY();
+	// clang-format on
+	if (error) {
+		throw duckdb::InternalException("(PGDuckDB/PostgresFunctionGuard) %s", edata->message);
```

Comment:
I don't think it makes sense to add this prefix, the error didn't really originate here, we are simply transforming it, so having a reference to the guard won't help you find where it originated. Especially if we'll also add conversion the other way around I can see many prefixes being added to a simple error message.
```suggestion
		throw duckdb::InternalException("%s", edata->message);
```